internal/kafka/internal/services/kafka.go

```diff
@@ -174,30 +253,36 @@ func (k *kafkaService) DetectInstanceType(kafkaRequest *dbapi.KafkaRequest) (typ
 		return types.STANDARD, nil
 	}
 
-	return types.EVAL, nil
+	return types.DEVELOPER, nil
 }
 
 // reserveQuota - reserves quota for the given kafka request. If a RHOSAK quota has been assigned, it will try to reserve RHOSAK quota, otherwise it will try with RHOSAKTrial
 func (k *kafkaService) reserveQuota(kafkaRequest *dbapi.KafkaRequest) (subscriptionId string, err *errors.ServiceError) {
-	if kafkaRequest.InstanceType == types.EVAL.String() {
-		if !k.kafkaConfig.Quota.AllowEvaluatorInstance {
-			return "", errors.NewWithCause(errors.ErrorForbidden, err, "kafka eval instances are not allowed")
+	if kafkaRequest.InstanceType == types.DEVELOPER.String() {
+		instType, err := k.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(kafkaRequest.InstanceType)
+
+		if err != nil {
+			return "", errors.NewWithCause(errors.ErrorGeneral, err, "unable to reserve quota")
 		}
 
-		// Only one EVAL instance is admitted. Let's check if the user already owns one
+		if !k.kafkaConfig.Quota.AllowDeveloperInstance {
+			return "", errors.NewWithCause(errors.ErrorForbidden, err, "kafka %s instances are not allowed", instType.DisplayName)
+		}
+
+		// Only one DEVELOPER instance is admitted. Let's check if the user already owns one
 		dbConn := k.connectionFactory.New()
 		var count int64
 		if err := dbConn.Model(&dbapi.KafkaRequest{}).
-			Where("instance_type = ?", types.EVAL).
+			Where("instance_type = ?", types.DEVELOPER).
 			Where("owner = ?", kafkaRequest.Owner).
 			Where("organisation_id = ?", kafkaRequest.OrganisationId).
 			Count(&count).
 			Error; err != nil {
-			return "", errors.NewWithCause(errors.ErrorGeneral, err, "failed to count kafka eval instances")
+			return "", errors.NewWithCause(errors.ErrorGeneral, err, "failed to count kafka %s instances", instType.DisplayName)
```

Comment:
would returning an DB error expose anything that shouldnt be exposed/returned. not sure where this error is consumed and what is done, just curious

Reply:
That totally depends on the error returned by the DB. I don't think in this case there should be anything sensible. It is performing a count on the database with information that is known when the kafka instance is created.